src/components/wallet/fetch-wallet-balance.ts

```diff
@@ -0,0 +1,72 @@
+import { tokenDetails } from './wallet-data'
+import {
+  BalanceType,
+  ParamsType,
+  ErrorType,
+  TokenDetailsType,
+  ConvertedBalanceType,
+  WalletBalanceType,
+} from './types'
+
+export const fetchTokenRate = async (
```

Comment:
Same thing with this. We can always reuse this. I think it's better we move it to the utils folder.